design/cdk-bootstrap.md

```diff
@@ -0,0 +1,199 @@
+# cdk-bootstrap
+
+`cdk-bootstrap` is a tool in the AWS CDK toolchain responsible for populating a given environment
+(that is, a combination of AWS account and region)
+with resources required by the CDK to perform deployments into that environment.
+It's a successor to the `cdk bootstrap` sub-command that is part of the `@aws-cdk/core` package,
+and is needed because of the increased bootstrapping requirements present in the
+["Continuous delivery for CDK apps"](https://github.com/aws/aws-cdk/pull/3437) design document.
+
+## Command-line interface
+
+Invoking the `bootstrap` tool looks something like this:
+
+```shell
+$ cdk-bootstrap [--trust accountId[,otherAccountId]...] aws:://account/region
+```
+
+If the `trust` option has been passed on the CLI,
+the tool will present the user with the following "menu":
+
+```shell
+This will create a new IAM Role, trusting the CloudFormation service principal,
+that will be used for deploying your CloudFormation stacks in this environment.
+Please choose the Managed Policy or Policies to attach to the Role
+that contain the permissions required to perform those deployments:
+
+[1]  AdministratorAccess
+[2]  AmazonDynamoDBFullAccess
+[3]  AmazonEC2FullAccess
+[4]  AmazonECS_FullAccess
+[5]  AmazonS3FullAccess
+[6]  AmazonSageMakerFullAccess
+[7]  AmazonSESFullAccess
+[8]  AmazonSNSFullAccess
+[9]  AmazonSQSFullAccess
+[10] AWSLambdaFullAccess
+[11] SimpleWorkflowFullAccess
+...
+
+Please input the number of the chosen Policy from the above list,
+or provide a fully qualified ARN of a different Managed Policy.
+You can input multiple numbers and/or ARNs (up to 10),
+separated by commas.
+
+>
+```
+
+After the user has made their choice,
+or if the `trust` option has not been passed,
+bootstrapping will proceed.
+
+### Bootstrap resources
+
```

Comment:
Please expand the entire bootstrap CloudFormation template here. It will be good way to ensure we didn't forget anything and that all the heuristics (like physical names) are expressible using simple CloudFormation and not require some fancy coding.

It's important to keep it like that because eventually we will want to support something like StackSets in order to bootstrap at scale, and so we need to protect ourselves from getting too fancy.

Reply:
Looking forward to seeing this template. This will eventually be the core of this design document.

Reply:
Done.